### PR TITLE
Ingredients UI(without api attached)

### LINKED ISSUE #10: Make UI for edit ingredient page (mobile)
As with all UI issues, just the components, no server/other page connection

### LINKED ISSUE #11: Make the UI for the ingredients page (mobile)
This issue is just for all the components/layout. We'll connect this page to other pages and to the server in subsequent issues.

### PR DESCRIPTION
**Pull request by smaliszewski15**
Creation of the Ingredients Page UI (without the API attached)

### CODE REVIEW

File: lib/utils/globals.dart [hunk 1]
```diff
@@ -1,8 +1,17 @@
...
 import 'package:smart_chef/utils/colors.dart';
 
 const double bottomIconSize = 55;
 const double bottomRowHeight = 90;
+const double topBarIconSize = 28;
+
+bool ingredientPage = true;
+bool recipePage = false;
+bool shoppingCartPage = false;
+bool userProfilePage = false;
```
Comment (nashirj) [inline lib/utils/globals.dart:14]: What are these bools used for?

File: lib/screens/UsersProfileScreen.dart [hunk 2]
```diff
@@ -33,6 +33,9 @@ class UserProfilePage extends StatefulWidget {
 class _UserProfilePageState extends State<UserProfilePage> {
   @override
   void initState() {
+    setState(() {
+      shoppingCartPage = true;
```
Comment (nashirj) [inline lib/screens/UsersProfileScreen.dart:37]: Why is shoppingCartPage set true when you're on this page?

File: lib/screens/IngredientScreen.dart [hunk 3]
```diff
@@ -618,3 +830,680 @@ class _IngredientPageState extends State<IngredientPage> {
...
+                ),
+              ),
+            ],
+          ),
+        ),
+      ),
+    );
+  }
+
+  //TODO(26): Allow searching of ingredients to add predefined things
```
Comment (nashirj) [inline lib/screens/IngredientScreen.dart]: Pls add space between `//` and start of comment e.g. `// TODO`

File: lib/screens/IngredientScreen.dart [hunk 4]
```diff
@@ -618,3 +830,680 @@ class _IngredientPageState extends State<IngredientPage> {
...
+                              style: const TextStyle(
+                                fontSize: 20,
+                                color: Colors.black,
+                              ),
+                              onChanged: (quantity) {
+                                if (quantity.isEmpty) {
+                                  setState(() => unfilledQuantity = true);
+                                } else {
+                                  unfilledQuantity = false;
+                                }
```
Comment (nashirj) [inline lib/screens/IngredientScreen.dart]: `setState(() => unfilledQuantity = quantity.isEmpty);`

File: lib/screens/IngredientScreen.dart [hunk 5]
```diff
@@ -618,3 +830,680 @@ class _IngredientPageState extends State<IngredientPage> {
...
+    );
+  }
+
+//TODO(17): Attach API to get specified ingredient information
+//
+// List fetchIngredient(int ID) {
+//   ingredientInfo = a json of some sort;
+// }
+//
+}
```
Comment (nashirj) [inline lib/screens/IngredientScreen.dart]: I think you have this function duplicated above, can remove one of the two

File: lib/screens/IngredientScreen.dart [hunk 6]
```diff
@@ -618,3 +830,680 @@ class _IngredientPageState extends State<IngredientPage> {
...
+                  ),
+                  backgroundColor: mainScheme,
+                  padding:
+                      const EdgeInsets.symmetric(vertical: 15, horizontal: 25),
+                  shadowColor: Colors.black,
+                ),
+                child: const Text(
+                  'Scan!',
+                  style: TextStyle(
+                    fontSize: 50,
```
Comment (nashirj) [inline lib/screens/IngredientScreen.dart:1391]: You've already done a good job of creating variables, but I would encourage you to extract all hard coded font/icon sizes into variables for two reasons:
1) easier to change across all your pages
2) encourages you to be intentional about choosing a new size that isn't used elsewhere already

**Review by nashirj — APPROVED** [review]

File: lib/utils/globals.dart (hunk 1 above)
Comment (smaliszewski15) [reply lib/utils/globals.dart:14]: i forgot to get rid of them. was gonna try to break the repetition of the bottom bar buttons but it ended up not working out.

File: lib/screens/IngredientScreen.dart (hunk 5 above)
Comment (smaliszewski15) [reply lib/screens/IngredientScreen.dart]: The functions were copy-pasted, but they are in two different classes. Their actual functionality will ultimately be different once they are complete, but i put them their as a reminder for myself

File: lib/screens/IngredientScreen.dart (hunk 6 above)
Comment (smaliszewski15) [reply lib/screens/IngredientScreen.dart:1391]: I will try to do this in the future, and will change all hardcoded stuff like that over time as I see it.

**Pull request merged**